cpp/dolfinx/fem/Expression.h

```diff
@@ -8,30 +8,34 @@
 
 #include "Constant.h"
 #include "Function.h"
-#include "pack.h"
-#include "utils.h"
 #include <algorithm>
 #include <array>
 #include <concepts>
 #include <dolfinx/common/types.h>
 #include <dolfinx/mesh/Mesh.h>
 #include <functional>
+#include <memory>
 #include <span>
 #include <utility>
 #include <vector>
 
 namespace dolfinx::fem
 {
-/// @brief Represents a mathematical expression evaluated at a
-/// pre-defined set of points on the reference cell.
+template <dolfinx::scalar T, std::floating_point U>
+class Function;
+
+/// @brief An Expression represents a mathematical expression evaluated
+/// at a pre-defined set of points on a reference cell.
 ///
-//// This class closely follows the concept of a UFC Expression.
+/// An Expression can be evaluated using ::assemble_expression.
```

Comment:
Wouldn't `tabulate_expression` be more appropriate? For me assembly indicates some summation into a tensor.